OWML.ModHelper/ModMergedConfig.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using OWML.Common;
+using OWML.Logging;
+
+namespace OWML.ModHelper
+{
+    public class ModMergedConfig : IModConfig
+    {
+        public bool Enabled { get => _userConfig.Enabled; set => _userConfig.Enabled = value; }
+        public Dictionary<string, object> Settings
+        {
+            get {
+                var settings = new Dictionary<string, object>(_defaultConfig.Settings);
+                _userConfig.Settings.ToList().ForEach(x => settings[x.Key] = x.Value);
+                return settings;
+            }
+            set => _userConfig.Settings = value;
+        }
+
+        private IModConfig _userConfig;
+        private readonly IModConfig _defaultConfig;
+        private readonly IModManifest _manifest;
+
+        public ModMergedConfig(IModConfig userConfig, IModConfig defaultConfig, IModManifest manifest)
+        {
+            _userConfig = userConfig;
+            _defaultConfig = defaultConfig;
+            _manifest = manifest;
+            FixConfigs();
+        }
+
+        public T GetSettingsValue<T>(string key)
+        {
+            if (_userConfig.Settings.ContainsKey(key))
+            {
+                return _userConfig.GetSettingsValue<T>(key);
+            }
+            return _defaultConfig.GetSettingsValue<T>(key);
+        }
+
+        public void SetSettingsValue(string key, object value)
+        {
+            _userConfig.SetSettingsValue(key, value);
+        }
+
+        public IModConfig Copy()
+        {
+            return new ModConfig
+            {
+                Enabled = Enabled,
+                Settings = new Dictionary<string, object>(Settings)
+            };
+        }
+
+        private void FixConfigs()
```

Comment:
All of the code starting with this method until the end of this class is just copied straight from ModData.cs, with minor adjustments to make it work here instead. This code would be greatly simplified after the next PR is done, which will make config.json have only values, and thus have a much simpler process for finding inconsistencies.